src/ol/proj/proj4.js

```diff
@@ -93,11 +93,45 @@ export function register(proj4) {
  * @return {Promise<string>} The proj4 definition.
  */
 let epsgLookup = async function (code) {
-  const response = await fetch(`https://epsg.io/${code}.proj4`);
+  const fromFile = async function (code) {
+    const response = await fetch(`https://epsg.io/${code}.proj4`);
+    if (!response.ok) {
+      throw new Error(`Unexpected response from epsg.io: ${response.status}`);
+    }
+    return response.text();
+  };
+
+  const response = await fetch(
+    `https://epsg.io/?format=json&trans=1&q=${code}`
+  );
   if (!response.ok) {
-    throw new Error(`Unexpected response from epsg.io: ${response.status}`);
+    return fromFile(`${code}`);
   }
-  return response.text();
+  return response.json().then((json) => {
+    const results = json['results'];
+    if (results?.length > 0) {
+      const result = results.filter(
+        (r) => r['authority'] === 'EPSG' && r['code'] === code.toString()
+      )[0];
+      if (result) {
+        const transforms = result['trans'];
+        if (transforms?.length > 0) {
+          const transform = transforms.filter(
+            (t) =>
+              t['grids_list']?.length === 0 &&
+              t['target_crs'] === 4326 &&
+              t['deprecated'] === 0 &&
+              t['usable'] === true
+          )[0]?.['code_trans'];
```

Comment:
You could sort the results by accuracy, and use the one with the highest accuracy instead of the first.